### PR TITLE
libfab_poller(): make sure the end-point is not freed while being used.

### PR DESCRIPTION
**Pull request by nikitadanilov**
See https://github.com/Seagate/cortx-motr/issues/1688:

libfab_poller()->libfab_txep_event_check(xep,
...)->libfab_buf_done()->libfab_buf_complete()->m0_net_buffer_event_post()->user-call-back.

The call-back frees the buffer and releases the last reference on the
end-point. The end-point is freed. On the return to libfab_txep_event_check()
the next iteration of m0_tl_teardown(fab_sndbuf, &txep->fep_sndbuf, fbp) loop
crashes.

Solution: acquire and release an additional reference to the end-point while it
is being processed by poller.

Signed-off-by: Nikita Danilov <nikita.danilov@seagate.com>

### CODE REVIEW

**Review by huanghua78 — APPROVED** [review]

**Review by upendrapatwardhan — APPROVED** [review]

**Comment by mehjoshi** [comment]
Waiting for @upendrapatwardhan to complete his analysis/review and give a +1 before merging.

**Pull request merged**
